file_diffs.py

```diff
@@ -234,6 +249,14 @@ def update_view(self, view, edit, tmp_file):
                 return
             view.replace(edit, region, self.get_content_from_file(tmp_file))
 
+    def file_will_be_read_from_disk(self, file):
+        view = self.view.window().find_open_file(file)
+        return bool(os.path.exists(file) and not (view and view.is_dirty()) and not (view and self.view_has_a_selection(view)))
```

Comment:
Strictly speaking you don't need the cast to `bool()`
```suggestion
        return os.path.exists(file) and not (view and view.is_dirty()) and not (view and self.view_has_a_selection(view))
```
No need to change the PR, I'm just pointing this out because you said python wasn't your first language.

Reply:
Agreed. That's left from a previous refactoring.